client.go

```diff
@@ -141,6 +169,22 @@ func (c *Client[Req, Res]) CallClientStream(ctx context.Context) *ClientStreamFo
 	}
 }
 
+// CallClientStream calls a client streaming procedure in simple mode.
```

Comment:
As a reader, I don't know what simple mode is (I'm not sure what it is in this context myself). This, and the other methods for simple mode, need Godoc explaining what the differences are, likely 3-6 sentences minimum, especially as most share the same function signature.